include/options.hxx

```diff
@@ -111,45 +155,178 @@ public:
   ///
   /// @param[in] parent        Parent object
   /// @param[in] sectionName   Name of the section, including path from the root
-  Options(Options *parent, string sectionName)
-      : parent(parent), sectionName(std::move(sectionName)){};
+  Options(Options *parent_instance, string full_name)
+      : parent_instance(parent_instance), full_name(std::move(full_name)){};
 
   /// Destructor
   ~Options();
 
-  /// Get a pointer to the only root instance (singleton)
-  static Options* getRoot();
-
-  /*!
-   * Free all memory
-   */
+  /// Get a reference to the only root instance
+  static Options &root();
+  
+  /// Free all memory
   static void cleanup();
 
-  // Setting options
-  template<typename T> void forceSet(const string &key, T t, const string &source=""){
-    set(key,t,source,true);
+  /// Get a sub-section or value
+  ///
+  /// Example:
+  ///
+  /// Options parent;
+  /// auto child  = parent["child"];
+  ///
+  /// parent is now a section.
+  Options& operator[](const string &name);
```

Comment:
Do we need a `const` version as well? If implemented, it would need to `throw` if a section/value didn't exist, instead of creating a new subsection. We'd probably also want a `const` version of `getSection` as well in that case.

---

A possible downside to this implementation is that this:

```cpp
int value = options["sub-section"];
```

is a runtime error, rather than a compile-time error, but I'm not sure there's much to be done about that.